File: monsoon3.0/src/main/java/wtf/monsoon/impl/module/movement/flight/VanillaFlight.java
package wtf.monsoon.impl.module.movement.flight;

import io.github.nevalackin.homoBus.Listener;
import io.github.nevalackin.homoBus.annotations.EventLink;
import wtf.monsoon.api.module.Module;
import wtf.monsoon.api.setting.ModeProcessor;
import wtf.monsoon.api.setting.Setting;
import wtf.monsoon.impl.event.EventPreMotion;

public class VanillaFlight extends ModeProcessor {

    private final Setting<Double> speedValue = new Setting<Double>("Speed", 0.5)
            .minimum(0.0D)
            .maximum(2.0D)
            .incrementation(0.05D)
            .describedBy("The speed you will go.");

    public VanillaFlight(Module parentModule) {
        super(parentModule);
    }

    @Override
    public void onDisable() {
        super.onDisable();
        mc.thePlayer.capabilities.isFlying = false;
    }

    @EventLink
    public final Listener<EventPreMotion> eventPreMotionListener = e -> {
        player.setSpeed(speedValue.getValue());
        mc.thePlayer.capabilities.isFlying = true;
    };

    @Override
    public Setting[] getModeSettings() {
        return new Setting[] { speedValue };
    }

}

File: monsoon-3.x-v2/src/main/java/wtf/monsoon/client/modules/movement/flight/VanillaFlight.kt
package wtf.monsoon.client.modules.movement.flight

import me.bush.eventbuskotlin.EventListener
import me.bush.eventbuskotlin.listener
import wtf.monsoon.backend.module.MulticlassModule
import wtf.monsoon.backend.module.mode.ModuleMode
import wtf.monsoon.client.event.EventPacket
import wtf.monsoon.client.event.EventPreMotion
import wtf.monsoon.client.modules.movement.Flight

class VanillaFlight(name: String, parent: MulticlassModule) : ModuleMode<Flight>(name, parent) {

    override fun disable() {
        super.disable()
        mc.thePlayer.capabilities.isFlying = false;
    }

    @EventListener
    val preMotion = fun(_: EventPreMotion) {
        mc.thePlayer.capabilities.isFlying = true;
    }

}